app/src/main/java/io/appium/settings/RecorderService.java

```diff
@@ -0,0 +1,138 @@
+    private static final String BASE = "io.appium.settings.recording";
+    public static final String ACTION_RECORDING_START = BASE + ".ACTION_START";
+    public static final String ACTION_RECORDING_STOP = BASE + ".ACTION_STOP";
+    public static final String ACTION_RECORDING_RESULT_CODE = "result_code";
+    public static final String ACTION_RECORDING_FILENAME = "recording_filename";
+
+    private static RecorderThread recorderThread;
+
+    public RecorderService() {
+        super();
+    }
+
+    @Override
+    public void onDestroy() {
+        Log.v(TAG, "onDestroy:");
+        if (recorderThread != null && recorderThread.isRecordingContinue()) {
+            recorderThread.stopRecording();
+        }
+        super.onDestroy();
+    }
+
+    @Nullable
+    @Override
+    public IBinder onBind(final Intent intent) {
+        return null;
+    }
+
+    @SuppressLint("NewApi")
+    @Override
+    public int onStartCommand(final Intent intent, final int flags, final int startId) {
+        Log.v(TAG, "onStartCommand:intent=" + intent);
+
+        int result = START_STICKY;
+        final String action = intent != null ? intent.getAction() : null;
+        Log.v(TAG, "onStartCommand:intent=" + "action string: " + action);
+        if (ACTION_RECORDING_START.equals(action)) {
+            showNotification(TAG); // TODO is this really necessary
+
+            MediaProjectionManager mMediaProjectionManager =
+                    (MediaProjectionManager) getSystemService(Context.MEDIA_PROJECTION_SERVICE);
+
+            startScreenRecord(mMediaProjectionManager, intent);
+        } else if (ACTION_RECORDING_STOP.equals(action)) {
+            Log.v(TAG, "onStartCommand:intent=" + "stop");
+            stopScreenRecord();
+            result = START_NOT_STICKY;
+        }
+
+        return result;
+    }
+
+    /**
+     * start screen recording
+     */
+    @SuppressLint("NewApi")
+    private void startScreenRecord(MediaProjectionManager mediaProjectionManager,
+                                   final Intent intent) {
+        if (recorderThread == null) {
+            Log.v(TAG, "recorder thread null");
+
+            final int resultCode = intent.getIntExtra(ACTION_RECORDING_RESULT_CODE, 0);
+            // get MediaProjection
+            final MediaProjection projection = mediaProjectionManager.getMediaProjection(resultCode,
+                    intent);
+            if (projection != null) {
+                Log.v(TAG, "projection different from null");
+                final DisplayMetrics metrics = getResources().getDisplayMetrics();
+                int width = metrics.widthPixels;
+                int height = metrics.heightPixels;
+                if (width > height) {
+                    // Horizontal
+                    final float scale_x = width / 1920f;
+                    final float scale_y = height / 1080f;
+                    final float scale = Math.max(scale_x,  scale_y);
```

Comment:
I can observe code duplication in both `if` branches

Reply:
done